src/Quasar/Advanced/Types.purs

```diff
@@ -234,16 +236,24 @@ instance decodeJsonGrantedTo ∷ DecodeJson GrantedTo where
 
     parseUserId ∷ String → Either String UserId
     parseUserId str =
-      Str.stripPrefix (Str.Pattern "user:") str # maybe (Left "Incorrect user") (pure <<< UserId)
+      Str.stripPrefix (Str.Pattern "user:") str
+      >>= fromString 
+      # map UserId
+      # note "Incorrect user"
+      
 
     parseTokenId ∷ String → Either String TokenId
     parseTokenId str =
-      Str.stripPrefix (Str.Pattern "token:") str # maybe (Left "Incorrect token") (pure <<< TokenId)
+      Str.stripPrefix (Str.Pattern "token:") str
+      >>= fromString 
+      # map TokenId
+      # note "Incorrect token"
+      
 
 parseGroup ∷ String → Either String GroupPath
 parseGroup string =
   Str.stripPrefix (Str.Pattern "group:") string
-  # maybe (Left "Incorrect group") pure
+  # note "Incorrect group"
```

Comment:
Let's change these "Incorrect <thing>" messages to "Could not parse <thing>" instead, while we're in here. We don't want a user to end up seeing these messages as they are, as it sounds like they're doing something wrong, when it's not at all something they have control over.